.ci/Jenkinsfile

```
@@ -322,139 +459,72 @@ def generateFunctionalTestStep(Map args = [:]){
+        withEnv(envContext) {
+          // This step will help to send the APM traces to the
+          // APM service defined by the Otel Jenkins plugin.
+          // TODO: This will need to be integrated into the provisioned VMs
+          withOtelEnv() {
+            // Start node, capture ip address
+            ansible("${env.WORKSPACE}",
+                    runId,
+                    "-t start-node --extra-vars=\"stackRunner=${stackRunner.ip} nodeLabel=${platform} nodeImage=${machine.image} nodeInstanceType=${machine.instance_type}\"")
 
-def grabWorkerIP(){
-  def linuxIp = ''
-  retryWithSleep(retries: 3, seconds: 5, backoff: true){
-    linuxIp = sh(label: 'Get IP', script: '''hostname -I | awk '{print $1}' ''', returnStdout: true)?.trim()
-    log(level: 'INFO', text: "Worker IP '${linuxIp}'")
-    if(!linuxIp?.trim()){
-      error('Unable to get the Linux worker IP')
-    }
-  }
-  return linuxIp
-}
+            def testRunner = getNodeIp("${env.WORKSPACE}",
+                                       platform)
 
+            // Configure node for testing
+            ansible("${env.WORKSPACE}",
+                    runId,
+                    "-i \"${testRunner.ip},\" -t setup-node --extra-vars=\"stackRunner=${stackRunner.ip} nodeLabel=${platform} nodeImage=${machine.image} nodeInstanceType=${machine.instance_type}\"")
 
-/**
-* Tear down the setup for the static workers.
-*/
-def tearDown(Map args = [:]){
-  catchError(buildResult: 'SUCCESS', stageResult: 'SUCCESS') {
-    dir("${BASE_DIR}"){
-      sh(label: 'Remove the entire module cache', script: 'go clean -modcache', returnStatus: true)
-    }
-    if (isStaticWorker(labels: args.labels)) {
-      dir("${WORKSPACE}") {
-        deleteDir()
+            sshexec("${env.WORKSPACE}",
+                    testRunner,
+                    """sudo bash /home/${testRunner.user}/e2e-testing/.ci/scripts/functional-test.sh "${tags}" """)
+          }
+        }
+      } finally {
+        def testRunner = getNodeIp("${env.WORKSPACE}",
+                                   platform)
+        sh "mkdir -p outputs/${testRunner.ip} || true"
+        sshexec("${env.WORKSPACE}",
+                testRunner,
+                """sudo chown ${testRunner.user}:${testRunner.user} -R /home/${testRunner.user}/e2e-testing/outputs """)
+        scpr("${env.WORKSPACE}",
+             testRunner,
+             "e2e-testing/outputs/TEST-*${runId}*.xml",
+             "outputs/${testRunner.ip}/.")
+        sh "ls -l outputs/${testRunner.ip}"
+        ansible("${env.WORKSPACE}",
+                runId,
+                "-t destroy --extra-vars=\"nodeLabel=${platform} nodeImage=${machine.image} nodeInstanceType=${machine.instance_type}\"")
+        junit allowEmptyResults: true,
+          keepLongStdio: true,
+          testResults: "outputs/${testRunner.ip}/TEST-*${runId}*.xml"
+        archiveArtifacts allowEmptyArchive: true,
+          artifacts: "outputs/${testRunner.ip}/TEST-*${runId}*.xml"
```

For clarity, let's use parenthesis for multiline steps
```suggestion
        junit(allowEmptyResults: true,
          keepLongStdio: true,
          testResults: "outputs/${testRunner.ip}/TEST-*${runId}*.xml")
        archiveArtifacts(allowEmptyArchive: true,
          artifacts: "outputs/${testRunner.ip}/TEST-*${runId}*.xml")
```